replace_and_make.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+shopt -s nullglob
+
+unreplaced_unrenamed_state=$(cat renames/unreplaced_unrenamed_state.txt 2>/dev/null)
+unreplaced_state=${unreplaced_unrenamed_state:0:1}
+unrenamed_state=${unreplaced_unrenamed_state:2:3}
+
+if [ "$unreplaced_state" = "1" ] || [ "$unrenamed_state" = "1" ] ; then
+	# Make an entire copy of the repo including ONLY files that are tracked and untracked, but not ignored
+	# as well as files which the user has to supply
+	set -e
+	echo "Finding files to copy over!"
+	unignored_files=$(git ls-files --exclude-standard --others --cached)
+	tools_bin_files=(tools/bin/*)
+	mwccarm_files=(tools/mwccarm/*/*/*.*)
+	supplied_files="${tools_bin_files[@]} ${mwccarm_files[@]} tools/mwccarm/license.dat sub/ARM7-TS.lcf.template ARM9-TS.lcf.template mwldarm.response.template"
+	files_to_copy="${unignored_files} ${supplied_files}"
+	echo "Copying over all modified files!"
+
+	rsync -auW --ignore-missing-args --files-from=<(echo "${files_to_copy}" | tr " " "\n") . rename_repo
+	cd rename_repo
+	unreplaced_unrenamed_state=$(cat renames/unreplaced_unrenamed_state.txt 2>/dev/null)
+	unreplaced_state=${unreplaced_unrenamed_state:0:1}
+	unrenamed_state=${unreplaced_unrenamed_state:2:3}
+	if [ "$unreplaced_state" = "1" ] ; then
+		python3 util/rereplace.py
```

Comment:
can we not introduce python as a requirement to build this repo please, it's fine having the occasional decomp tool in it, but *please* put this in C if it's needed to build

Reply:
Python is preferable because the script will likely need changes in its algorithm as the repository evolves. It also isn't required for building if you never use the scripts in the first place.

I can always wait for pokeheartgold to accept meson anyway, then Python will be a requirement :trollface:

Reply:
this is also true of most, if not all c tools, and they're fine as is, I'd rather just not make python necessary at this moment

Reply:
It's not really a true requirement in that the repo can get to a state where the python dependency can be removed. It's just something temporary while the repo is being worked on to make reviewing PRs much easier. It can always be added as a contributor-only requirement in the INSTALL.md

Reply:
> I can always wait for pokeheartgold to accept meson anyway, then Python will be a requirement :trollface:

While I strongly recommend meson wherever possible, please do note that there *is* a c99 implementation of it: https://sr.ht/~lattis/muon

Reply:
Nice to see you here :)
 
> > I can always wait for pokeheartgold to accept meson anyway, then Python will be a requirement :trollface:
> 
> While I strongly recommend meson wherever possible, please do note that there _is_ a c99 implementation of it: https://sr.ht/~lattis/muon

That would require adding metrowerks support to muon as well, not to mention muon lacks some meson features that the build system at [https://github.com/pret/pokeplatinum](url) relies on, and would be needed in a meson port of this repo too.